addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/RemoteControllerWebSocket.java

```diff
@@ -0,0 +1,678 @@
+
+    WebSocketRemote webSocketRemote = new WebSocketRemote();
+    WebSocketArt webSocketArt = new WebSocketArt();
+    WebSocketV2 webSocketV2 = new WebSocketV2();
+
+    /**
+     * Create and initialize remote controller instance.
+     *
+     * @param host                    Host name of the Samsung TV.
+     * @param port                    TCP port of the remote controller protocol.
+     * @param appName                 Application name used to send key codes.
+     * @param uniqueId                Unique Id used to send key codes.
+     * @param remoteControllerService
+     */
+
+    public RemoteControllerWebSocket(String host, int port, String appName, String uniqueId,
+            RemoteControllerWebsocketCallback remoteControllerWebsocketCallback) {
+        super(host, port, appName, uniqueId);
+        this.client = new WebSocketClient();
+        this.client.addLifeCycleListener(this);
+
+        this.callback = remoteControllerWebsocketCallback;
+    }
+
+    private boolean isConnected() {
+        return webSocketRemote.isConnected();
+    }
+
+    @Override
+    public void openConnection() throws RemoteControllerException {
+        if (!(client.isStarted() || client.isStarting())) {
+            logger.debug("RemoteControllerWebSocket start Client");
+            try {
+                client.start();
+                client.setMaxBinaryMessageBufferSize(1000000);
+                // websocket connect will be done in lifetime handler
+                return;
+
+            } catch (Exception e) {
+                logger.warn("Cannot connect to websocket remote control interface: " + e.getMessage());
+                throw new RemoteControllerException(e);
+            }
+        }
+        connectWebSockets();
+    }
+
+    private void connectWebSockets() {
+        try {
+            webSocketRemote.connect(new URI("ws", null, host, port, WS_ENDPOINT_REMOTE_CONTROL, "name=openhab", null));
+        } catch (RemoteControllerException | URISyntaxException e) {
+            logger.warn("Problem connecting to remote websocket", e);
+        }
+        try {
+            webSocketArt.connect(new URI("ws", null, host, port, WS_ENDPOINT_ART, "name=openhab", null));
+        } catch (RemoteControllerException | URISyntaxException e) {
+            logger.warn("Problem connecting to artmode websocket", e);
+        }
+        try {
+            webSocketV2.connect(new URI("ws", null, host, port, WS_ENDPOINT_V2, "name=openhab", null));
+        } catch (RemoteControllerException | URISyntaxException e) {
+            logger.warn("Problem connecting to V2 websocket", e);
+        }
+    }
+
+    private void closeConnection() throws RemoteControllerException {
+        logger.debug("RemoteControllerWebSocket closeConnection");
+
+        try {
+            webSocketRemote.close();
+            webSocketArt.close();
+            webSocketV2.close();
+            client.stop();
+        } catch (Exception e) {

```

Comment:
Generally speaking, catching `Exception` should be avoided. Can you be more explicit on what exception you are expecting here?

Reply:
Unfortunately the WebSocketClient documentation is not more explicit. I have limited to scope of the try to only this client.stop() now.